index.html

```diff
@@ -977,13 +1003,50 @@
 				html = html + "<span class='varname'>Institute: </span> <span class='attribute'>" + data.institute + "</span>";
 				html = html + "<br />"
 				html = html + "<span class='varname'>Campus: </span> <span class='attribute'>" + standardizeCase(data.campus) + "</span>";
-				html = html + "<br />"
-				html = html + "<span class='varname'>School District: </span> <span class='attribute'>" + standardizeCase(data.district) + "</span>";
+				if (data.district) {
+					html = html + "<br />"
+					html = html + "<span class='varname'>School District: </span> <span class='attribute'>" + standardizeCase(data.district) + "</span>";
+				}
 				return html;
 				//this will return the string to the calling function
 
 			}
 
+// These are the popups for the polygon district layers, using Eldan's House/Senate 'show' logic
```

Comment:
Most of the rewriting I've done here is actually incidental to the bug - I just realised that there was repeated code, and straightened that out for ease of future maintenance.